source/hash_table.c

```diff
@@ -646,6 +646,37 @@ int aws_hash_table_foreach(
     return AWS_OP_SUCCESS;
 }
 
+bool aws_hash_table_eq(
+    bool (*value_eq)(const void *a, const void *b),
+    const struct aws_hash_table *a,
+    const struct aws_hash_table *b) {
+    if (aws_hash_table_get_entry_count(a) != aws_hash_table_get_entry_count(b)) {
```

Comment:
Do we want to add a check for the case when hash_table's are NULL?

Reply:
Generally, we assume that arguments should be non-null except for destructors and for methods that document themselves as accepting null. I'm not sure that null == null is a condition we want to explicitly support here.

Reply:
I see, thanks for stating the assumption.